frontend/src/components/UserProfile.js

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import styled from "styled-components";
+import { rgba } from "polished";
+
+import { user } from "../reducers/user";
+import { SubmitButton } from "./SubmitButton";
+import { Home } from "./Home";
+
+const URL = "https://project-auth-liza-kat.herokuapp.com/users";
+
+export const UserProfile = () => {
+  const dispatch = useDispatch();
+  const accessToken = useSelector((store) => store.user.login.accessToken);
+  const userId = useSelector((store) => store.user.login.userId);
+  const secretMessage = useSelector((store) => store.user.login.secretMessage);
+  const [loggedOut, setLoggedOut] = useState(false);
+
+  const handleLogout = () => {
+    dispatch(user.actions.logout());
+    setLoggedOut(true);
+  };
+  const testProfile = () => {
+    // Include userId in the path
+    fetch(`${URL}/${userId}/secret`, {
+      method: "GET",
+      // Include the accessToken to get the protected endpoint
+      headers: { Authorization: accessToken },
+    })
+      .then((res) => {
+        if (!res.ok) {
+          try {
+            throw new Error("error");
+          } catch (e) {
+            throw e;
+          }
+        }
+        return res.json();
+      })
+      .then((json) => {
+        dispatch(user.actions.setSecret({ secretMessage: json.secretMessage }));
+      })
+      .catch((err) => alert(err));
+  };
+
+  return (
+    <>
+      {loggedOut === false ? (
+        <Image>
+          <Container>
+            <Text>User's Profile</Text>
+            <Text>
+              {" "}
+              Hello, user. Reveal your secret by pressing the Test button
+            </Text>
+          </Container>
+          <SubmitButton title="Logout" onClick={handleLogout}></SubmitButton>
+          <SubmitButton title="Test" onClick={testProfile}></SubmitButton>
+          <Text1>{`${secretMessage}`}</Text1>
+        </Image>
+      ) : (
+        <Home />
+      )}
+    </>
+  );
+};
+
+const Image = styled.main`
+  background-image: url("${process.env.PUBLIC_URL + "/flower.jpg"}");
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  background-size: cover;
+`;
```

Comment:
If you are using the same styling in more components than one, like this one, maybe you can move the styling out from the component and then import this image styling 😍